packages/compiler/src/core/name-resolver.ts

```diff
@@ -190,6 +199,13 @@ export function createResolver(program: Program): NameResolver {
   const nullSym = createSymbol(undefined, "null", SymbolFlags.None);
   const augmentDecoratorsForSym = new Map<Sym, AugmentDecoratorStatementNode[]>();
 
+  /**
+   * Tracking the template parameters that are used.
+   */
+  const usedTemplateParameterDeclarationNodes = new Set<TemplateParameterDeclarationNode>();
```

Comment:
this doesn't belong here at this point if its only used in the checker

Reply:
remove the usedTemplateParameter list in check.

Reply:
no i mean this whole change doesn't belong in this file

Reply:
this is still not fixed, there should not be any change to this file